CPAC/pipeline/cpac_pipeline.py

```diff
@@ -693,67 +661,92 @@ def build_workflow(subject_id, sub_dict, c, pipeline_name=None, num_ants_cores=1
-        ("anat", "PRIORS_GRAY"),
-        ("anat", "PRIORS_WHITE"),
-        ("other", "configFileTwomm"),
-        ("anat", "template_based_segmentation_CSF"),
-        ("anat", "template_based_segmentation_GRAY"),
-        ("anat", "template_based_segmentation_WHITE"),
-        ("anat", "template_based_segmentation_WHITE"),
-        ("anat", "acpc_template_skull"),
-        ("anat", "acpc_template_brain"),
+        ("anat", ["network_centrality", "template_specification_file"]),
+        ("anat", ["nuisance_corrections", "2-nuisance_regression", "lateral_ventricles_mask"]),
+        ("anat", ["anatomical_preproc", "segmentation_workflow", "2-use_priors", "CSF_path"]),
+        ("anat", ["anatomical_preproc", "segmentation_workflow", "2-use_priors", "GM_path"]),
+        ("anat", ["anatomical_preproc", "segmentation_workflow", "2-use_priors", "WM_path"]),
+        ("anat", ["anatomical_preproc", "segmentation_workflow", "1-segmentation", "Template_Based", "CSF"]),
+        ("anat", ["anatomical_preproc", "segmentation_workflow", "1-segmentation", "Template_Based", "GRAY"]),
+        ("anat", ["anatomical_preproc", "segmentation_workflow", "1-segmentation", "Template_Based", "WHITE"]),
+        ("anat", ["anatomical_preproc", "acpc_alignment", "template_skull"]),
+        ("anat", ["anatomical_preproc", "acpc_alignment", "template_brain"]),
+        ("other", ["voxel_mirrored_homotopic_connectivity", "symmetric_registration", "FNIRT_pipelines", "config_file"]),
     ]
 
+    def get_nested_attr(c, template_key):
+        attr = getattr(c, template_key[0])
+        keys = template_key[1:]
+        def _get_nested(attr, keys):
+            if len(keys) > 1:
+                return(_get_nested(attr[keys[0]], keys[1:]))
+            elif len(keys):
+                return(attr[keys[0]])
+            else:
+                return(attr)
+        return(_get_nested(attr, keys))
+
+    def set_nested_attr(c, template_key, value):
+        attr = getattr(c, template_key[0])
+        keys = template_key[1:]
+        def _set_nested(attr, keys):
+            if len(keys) > 1:
+                return(_set_nested(attr[keys[0]], keys[1:]))
+            elif len(keys):
+                attr[keys[0]]=value
+            else:
+                return(attr)
+        return(_set_nested(attr, keys))
+
     for key_type, key in template_keys:
+        
+        attr = get_nested_attr(c, key)
 
-        if isinstance(getattr(c, key), str) or getattr(c, key) == None:
+        if isinstance(attr, str) or attr == None:
             
             node = create_check_for_s3_node(
-                key,
-                getattr(c, key), key_type,
-                input_creds_path, c.workingDirectory, map_node=False
+                key[-1],
+                attr, key_type,
+                input_creds_path, c.pipeline_setup['working_directory']['path'], map_node=False
             )
 
-            setattr(c, key, node)
-    
+            set_nested_attr(c, key, node)
```

Comment:
If you do 

```Python
c.set_nested(c, key, node)
```

or

```Python
c[key] = node
```

here, then you don't need to define `set_nested_attr` at all.